processor/processor.go

```diff
@@ -114,54 +174,52 @@ func (p *Processor) isError(e ast.Field) bool {
 
 func (p *Processor) Process(fset *token.FileSet, file *ast.File) error {
 	var patches []patch
+	var funcLitCond *funcTypeConditions
 
 	astutil.Apply(file, nil, func(c *astutil.Cursor) bool {
 		if c == nil {
 			return true
 		}
 
-		fn, ok := c.Node().(*ast.FuncDecl)
-		if !ok || fn == nil {
-			return true
-		}
+		p.packageName(c)
 
-		fname := p.functionName(*fn)
-		if !p.FunctionSelector.AcceptFunction(fname) {
-			return true
-		}
+		switch c.Node().(type) {
+		// original
+		case *ast.FuncDecl:
+			fnc, _ := c.Node().(*ast.FuncDecl)
+			funcName := p.functionName(*fnc)
+			if !p.FunctionSelector.AcceptFunction(funcName) {
+				return true
+			}
 
-		spanName := p.SpanName(p.methodReceiverTypeName(*fn), fname)
+			receiverName := p.methodReceiverTypeName(*fnc)
+			if receiverName == "" {
+				receiverName = p.PackageName
+			}
 
-		hasContext := false
-		hasError := false
+			spanName := p.SpanName(receiverName, funcName)
+			funcCond := &funcTypeConditions{Type: fnc.Type}
+			p.hasPrefixConditions(funcCond)
 
-		if t := fn.Type; t != nil {
-			if ps := t.Params; ps != nil {
-				for _, q := range ps.List {
-					if q == nil {
-						continue
-					}
-					hasContext = hasContext || p.isContext(*q)
-				}
+			if funcLit := functionLiteral(fnc, funcLitCond); funcLit != nil {
+				ps := p.Instrumenter.PrefixStatements(spanName, funcLitCond.hasError)
+				patches = append(patches, patch{pos: funcLit.Body.Pos(), stmts: ps})
 			}
 
-			if rs := t.Results; rs != nil {
-				for _, q := range rs.List {
-					if q == nil {
-						continue
-					}
-					hasError = hasError || p.isError(*q)
-				}
+			if !funcCond.hasContext {
+				return true
 			}
-		}
 
-		if !hasContext {
+			ps := p.Instrumenter.PrefixStatements(spanName, funcCond.hasError)
+			patches = append(patches, patch{pos: fnc.Body.Pos(), stmts: ps})
+		// anonymous
+		case *ast.FuncLit:
+			fnc, _ := c.Node().(*ast.FuncLit)
+			funcLitCond = &funcTypeConditions{Type: fnc.Type}
+			p.hasPrefixConditions(funcLitCond)
```

Comment:
better to move out to function
```suggestion
		case *ast.FuncLit:
		        p.processAnonymousFunction(c.Node().(*ast.FuncLit))
```